Doc/library/datetime.rst

```diff
@@ -2028,6 +2028,10 @@ For :class:`date` objects, the format codes for hours, minutes, seconds, and
 microseconds should not be used, as :class:`date` objects have no such
 values.  If they're used anyway, ``0`` is substituted for them.
 
+For the :meth:`datetime.strptime` class method, the default value is ``1900-01-01T00:00:00.000``
+and when using :meth:`strptime`, any components requested in the format string but
+missing from the object itself will be pulled from the default value. [#]_
```

Comment:
The "missing from the object itself" still doesn't make sense here, there's no object, what this is documenting is what happens when you do this:

```python
datetime.strptime("12:00:00", "%H:%M:%S")
```

Since the primary `datetime` constructor has required arguments, and `0` is not a valid value for year, month or day, we need to specify the default value for missing components.

So instead of "any components requested in the format string but missing from the object itself will be pulled from the default value", I think it should be "components not specified in the format string will be pulled from the default value."

The only reason I say "I think" is that people *may* get confused between the situation where neither the value string nor the format string contains the required components (which is what we are talking about) and the situation where the format string specifies a component but it's missing from the value string. Hopefully my suggested wording makes it clear that we're talking about the first case.